modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQBrokerDetails.java

```diff
@@ -24,13 +24,39 @@
 import java.net.URISyntaxException;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import org.wso2.andes.jms.BrokerDetails;
 import org.wso2.andes.url.URLHelper;
 import org.wso2.andes.url.URLSyntaxException;
 
 public class AMQBrokerDetails implements BrokerDetails
 {
+
+    /**
+     * URL pattern
+     */
+    private static final Pattern URL_PATTERN = Pattern.compile("([a-z]+://.*)|([a-z]+:/.*)");
+    /**
+     * Password pattern in JMS URL
+     */
+    private static final Pattern PASSWORD_PATTERN = Pattern.compile("transport.jms.Password=([^/&]+)(&|$)");
+
+    private static final Pattern KEYSTORE_PWD_PATTERN = Pattern.compile("key_store_password=([^/&]+)(&|$)");
+    private static final Pattern TRUSTSTORE_PWD_PATTERN = Pattern.compile("trust_store_password=([^/&]+)(&|$)");
+    /**
+     * Security credential pattern in JMS URL
+     */
+    private static final Pattern SECURITY_CREDENTIAL_PATTERN = Pattern
+            .compile("java.naming.security.credentials=" + "([^/&]+)(&|$)");
```

Comment:
Do we need to have string concat here?

Reply:
Fixed in https://github.com/wso2/andes/pull/1035